x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/multivalue/MvPSeriesWeightedSum.java

```diff
@@ -89,14 +90,18 @@ protected TypeResolution resolveType() {
             return resolution;
         }
 
-        resolution = TypeResolutions.isType(p, dt -> dt == DOUBLE, sourceText(), SECOND, "double")
-            .and(isNotNullAndFoldable(p, sourceText(), SECOND));
-
+        resolution = TypeResolutions.isType(p, dt -> dt == DOUBLE, sourceText(), SECOND, "double");
         if (resolution.unresolved()) {
             return resolution;
         }
 
-        return resolution;
+        if (p.dataType() == NULL) {
+            // If the type is `null` this parameter doesn't have to be foldable. It's effectively foldable anyway.
+            // TODO figure out if the tests are wrong here, or if null is really different from foldable null
+            return resolution;
+        }
```

Comment:
FYI: if it's null, FoldNull rule will convert the full expression into a literal null. And we explicitly call that rule in the tests. See:

https://github.com/elastic/elasticsearch/blob/c5da25754007af51029a676a8784b16be16d8995/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/AbstractScalarFunctionTestCase.java#L93

_FoldNull checks if it's either NULL or is foldable and fold() == null. isFoldable() just checks foldability._

Btw, before touching anymore this method, I really think we should make it work with non-constants. Looks like we're "repeatedly" bumping into problems other functions don't have just because of it :')